core/app/models/spree/promotion/actions/free_shipping.rb

```diff
@@ -8,20 +8,21 @@ def perform(payload = {})
           order = payload[:order]
           promotion_code = payload[:promotion_code]
 
-          results = order.shipments.map do |shipment|
-            return false if promotion_credit_exists?(shipment)
-            shipment.adjustments.create!(
-              order: shipment.order,
-              amount: compute_amount(shipment),
-              source: self,
-              promotion_code: promotion_code,
-              label: label
-            )
-            true
+          created_adjustments = order.shipments.map do |shipment|
+            if !promotion_credit_exists?(shipment)
+              shipment.adjustments.create!(
+                order: shipment.order,
+                amount: compute_amount(shipment),
+                source: self,
+                promotion_code: promotion_code,
+                label: label
+              )
+            end
```

Comment:
Up to you which you think is better, but I probably would have switched this to use `next` to avoid unnecessary code churn.

```suggestion
          created_adjustments = order.shipments.map do |shipment|
            next if promotion_credit_exists?(shipment)
            
            shipment.adjustments.create!(
              order: shipment.order,
              amount: compute_amount(shipment),
              source: self,
              promotion_code: promotion_code,
              label: label
            )
```

Reply:
👍